predicators/spot_utils/perception_utils.py

```diff
@@ -377,18 +387,23 @@ def get_object_locations_with_detic_sam(
     res_segment = query_detic_sam(image_in=rotated_rgb,
                                   classes=classes,
                                   viz=plot)
-    if res_segment is None:
-        return []
 
-    # Detect multiple objects with their masks
-    obj_num = len(res_segment['masks'])
-    res_locations = []
-    for i in range(obj_num):
+    if len(res_segment['classes']) == 0:
+        return {}
+
+    ret_obj_positions: Dict[str, Tuple[float, float, float]] = {}
+    for i, obj_class in enumerate(res_segment['classes']):
+        # Check that this particular class is one of the
```

Comment:
is it possible that these assertions will crash? should we do something soft that doesn't crash instead?

Reply:
no - it shouldn't be possible